src/data/finset/basic.lean

```diff
@@ -654,18 +654,21 @@ theorem induction_on' {α : Type*} {p : finset α → Prop} [decidable_eq α]
   let ⟨hS, sS⟩ := finset.insert_subset.1 hs in h₂ hS sS has (hqs sS)) (finset.subset.refl S)
 
 /-- To prove a proposition about a nonempty `s : finset α`, it suffices to show it holds for all
-singletons and that if it holds for `t : finset α`, then it also holds for the `finset` obtained by
-inserting an element in `t`. -/
+singletons and that if it holds for nonempty `t : finset α`, then it also holds for the `finset`
+obtained by inserting an element in `t`. -/
 @[elab_as_eliminator]
-lemma nonempty.cons_induction {α : Type*} {s : finset α} (hs : s.nonempty) {p : finset α → Prop}
-  (h₀ : ∀ a, p {a}) (h₁ : ∀ ⦃a⦄ s (h : a ∉ s), p s → p (finset.cons a s h)) :
-  p s :=
+lemma nonempty.cons_induction {α : Type*}
+  {p : Π s : finset α, s.nonempty → Prop}
+  (h₀ : ∀ a, p {a} (singleton_nonempty _))
+  (h₁ : ∀ ⦃a⦄ (s : finset α) (h : a ∉ s) (hs : s.nonempty),
+    p s hs → p (finset.cons a s h) (nonempty_cons h)) {s : finset α} (hs : s.nonempty) :
```

Comment:
Since the type of `s` is explicit now, can we use dot notation on the cons? It would also be nice if `h₁` fits on one line, or at least if the later arguments were on a different line, I think it's a little tricky to read right now.

Reply:
Sure, I can try and do that and line wrap tomorrow - or feel free to push that cleanup yourself

Reply:
Alright, adjusted to fit on one line - I decided against `s.cons a` since it sort of feels backwards for `cons` which is confusing.